### DISCUSSION TITLE
MPF & MPF-MC 0.56 release thread

### DISCUSSION THREAD

**Discussion by toomanybrians** [missionpinball/missionpinball#57] [General]
It's been about 18 months since we've done a release, so I'd like to move 0.56 out of dev and release that, and move dev to 0.57.

I feel like the new installers have generally been ok. Looking ahead, we have some big changes needed in the next few months which I'd like to push to 0.57, including:

* Refactoring the asyncio use to support Python 3.10+
* Updating the ruamel.yaml interface to use their newest 0.17.x. We use 0.15.42 because we have customized it since the MPF configs diverge from pure YAML, and that's the latest version that works with our customizations. Those customizations need to be pulled forward into the latest ruamel.yaml which are also needed to support newer versions of Python.

I'm personally also in the process of updating the FAST platform interface to refactor and update it for the latest FAST stuff. I have those in a fork at the moment, but I figure I can merge those back into the mainline MPF tree as part of 0.57 too. So that would hopefully mean 0.57 can target the latest versions of Python as its main reason for existing.

So looking at the current 0.56/dev, I know we need to figure out the open issue of [MPF-MC audio not working on Windows](https://github.com/missionpinball/mpf-mc/issues/436), but apart from that, any reasons, or anything else to add to the list for 0.56 before we release it and start on 0.57?

Todo list before release:

- [x] Fix MPF-MC audio for Windows
- [x] Merge in [MPF PR 1634]([https://github.com/missionpinball/mpf/pull/1634](https://github.com/orgs/missionpinball/discussions/url))

**Comment by borgdog** [comment]
It would be nice to see the open pull request for chime sync ([https://github.com/missionpinball/mpf/pull/1634](https://github.com/missionpinball/mpf/pull/1634)) get finalized and added in.

**Reply by toomanybrians** [reply]
ok I'll take a look at that.

**Comment by toomanybrians** [comment]
Update to the audio not working in MPF-MC [issue](https://github.com/missionpinball/mpf-mc/issues/436): We determined that if we build the Windows wheel locally, it works. So the audio errors only happen with the wheel that's built by the GitHub runners.

So as a workaround, we can manually build the 0.56 release wheel. Luckily the MPF-MC doesn't change too often, so just getting this built once should be all that people need.

**Comment by cobra18t** [comment]
This is great, Thank you!

**Comment by toomanybrians** [comment]
Ok, this is done!

I have done 0.56 releases for MPF, MPF-MC, MPF Monitor, and the docs.

MPF-MC is comically 0.56.1 already 🤦‍♂️ since the GitHub process built the broken Windows packages and PyPI's security doesn't let you swap out the binary for an existing version. (Which, ugg, but also, yeah, that's a good feature.) So I had to update the MPF-MC actions to not build for Windows so I could manually upload the locally built one.

But the docs are updated for the new installs, etc, and everything seems to be good.

Personally I will be working on MPF 0.57 now to get through those points I mentioned in the first post. I do not expect that I will do anything with MPF-MC or MPF Monitor for a while. I would still love to have more help with MPF maintenance (boring, I know!), and in general I will be acting in a maintenance support mode as I don't have the time or interest to be the primary driver for everything. But I think we're in ok shape overall now.

**Comment by atummons** [comment]
Brian, thanks for all your work! I know a while back you asked about getting to 1.0 and people threw out ideas. e.g., consistent terminology throughout, a couple extra features. Is there an official list in issues or milestones to track towards?  My Python knowledge isn't great, but Jan helped me understand quite a bit. I can work toward some cleanup pieces (time permitting, so it might be slowly) if we have them lined out somewhere.

**Comment by worldpeace-germany** [comment]
Kind of the same here. Some python knowledge, but not too great. If there are some tasks you can hand over to someone with that skill set I am happy to help. Of course time limited.... Feel free to get in contact with me.
